Bio/Seq.py

```diff
@@ -852,23 +852,43 @@ def complement(self):
         Note in the above example, ambiguous character D denotes
         G, A or T so its complement is H (for C, T or A).
 
-        Trying to complement a protein sequence raises an exception.
+        Note that if the sequence contains neither T nor U, we
+        assume it is DNA and map any A character to T:
+
+        >>> Seq("CGA").complement()
+        Seq('GCT')
+        >>> Seq("CGAT").complement()
+        Seq('GCTA')
+        >>> Seq("CGAU").complement()
+        Seq('GCUA')
+
+        If you are dealing with RNA which might have no U, use:
+
+        >>> Seq("CGA").complement().transcribe()
+        Seq('GCU')
+
+        If the sequence contains both T and U, an exception is
+        raised:
+
+        >>> Seq("CGAUT").complement()
+        Traceback (most recent call last):
+           ...
+        ValueError: Mixed RNA/DNA found
+
+        Trying to complement a protein sequence gives a meaningless
+        sequence:
 
         >>> from Bio.Alphabet import generic_protein
         >>> my_protein = Seq("MAIVMGR", generic_protein)
         >>> my_protein.complement()
-        Traceback (most recent call last):
-           ...
-        ValueError: Proteins do not have complements!
+        Seq('KTIBKCY')
+
+        Here "M" was interpretted as the IUPAC ambiguity code for
+        "A" or "C", with complement "K" for "T" or "G". Likewise
+        "A" has complement "T". The letter "I" has not defined
+        meaning under the IUPAC convention, and is unchanged.
```

Comment:
I think the presence of letter "I" should raise a `ValueError`

Reply:
I am sympathetic, but where/how do we draw the line? Error on any ASCII letter other than the IUPAC code? I've seen X used like N though...

What about punctuation? The minus sign is an obvious one as a gap, but people will use all sorts of things in their own code - and historically Biopython accepted this.

Reply:
OK then let's leave it as is, without a `ValueError`.